bothan-binance/src/worker.rs

```diff
@@ -0,0 +1,75 @@
+use tokio::sync::mpsc::Sender;
+use tracing::error;
+
+use bothan_core::store::Store;
+use bothan_core::worker::{AssetStatus, AssetWorker, Error};
+
+use crate::api::websocket::BinanceWebSocketConnector;
+
+mod asset_worker;
+pub mod builder;
+pub(crate) mod error;
+mod types;
+
+/// A worker that fetches and stores the asset information from Binance's API.
+pub struct BinanceWorker {
+    connector: BinanceWebSocketConnector,
+    store: Store,
```

Comment:
In the future this will likely become an `Arc` since we plan to share store across multiple workers. Can change it now or leave it to future PRs.